Units/simple-elixir.d/input.ex

```diff
@@ -1,18 +1,190 @@
+# Most of this code is taken from the official Elixir documentation.
+# With comments and modifications
+# by Ivan Gonzalez Polanco <ivan14polanco@gmail.com>
 #
-# Taken from https://elixir-lang.org/
+# This code doesn't compile, since it's supposed to test de ctags elixir
+# parser, not the elixir elixir parser.
+
+#
+# f functions (def ...)
+#
+defmodule MyString do
+  def one_liner_func, do: :baz
+
+  def func_no_params do
+    #
+  end
+
+  # Function head
+  def join(string1, string2 \\ nil, separator \\ " ")
+
+  # Function with 1 arity
+  def join(string1, nil, _separator) do
+    private_function(string1)
+  end
+
+  # Normal function
+  def join(string1, string2, separator) do
+    string1 <> separator <> string2
+  end
+
+  # Private function
+  defp private_func(a), do: a <> " alone"
+
+  defp private_func_no_params do
+    #
+  end
+end
+
+#
+# c callbacks (defcallback ...)
+#
+defmodule URI.MyParser do
+  use Behaviour
+  # This is the new callback syntax
+  @callback default_port() :: integer
+
+  # This is the old (deprecated) callback syntax
+  defcallback parse(uri_info :: URI.t()) :: URI.t()
+end
+
+#
+# d delegates (defdelegate ...)
+#
+defmodule MyList do
+  defdelegate reverse(list), to: Enum
+  defdelegate other_reverse(list), to: Enum, as: :reverse
+end
+
+# TODO: I don't know how to identificate/differenciate an exception since they
+# don't _actually_ have name. Instead are defined like a struct.
+#
+# e exceptions      (defexception ...)
+#
+# defmodule MyAppError do
+#   defexception [:message]
+#
+#   @impl true
+#   def exception(value) do
```

Comment:
An anonymous tag is needed here.  ctags main part should provide such a feature. 
This is my job.

Reply:
See #2025.